docs/reference/search/search-your-data/learning-to-rank-model-training.asciidoc

```diff
@@ -0,0 +1,150 @@
+===== Extracting features for training
+
+One of the most important steps of the training process is to build the dataset that will be used by extracting and adding features to it. Eland provides another helper class, `eland.ml.ltr.FeatureLogger`, to aid in this process:
+
+[source,python]
+----
+from eland.ml.ltr import FeatureLogger
+
+# Create a feature logger that will be used to query {es} to retrieve the features:
+feature_logger = FeatureLogger(es_client, MOVIE_INDEX, ltr_config)
+----
+// NOTCONSOLE
+
+The FeatureLogger provides an `extract_features` method allowing you to extract features for a list of specific documents from your judgment list. At the same time, query parameters used by the feature extractors defined earlier can be passed:
+
+[source,python]
+----
+feature_logger.extract_features(
+    query_params:{"query": "foo"},
+    doc_ids=["doc-1", "doc-2"]
+)
+----
+// NOTCONSOLE
+
+Our https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/08-learning-to-rank.ipynb[example notebook^] provides a complete example explaining how to use the `FeatureLogger` to add features to the judgment list in order to build the training dataset.
+
+[discrete]
+[[learning-to-rank-model-training-feature-extraction-notes]]
+====== Notes on features extraction
+
+* We strongly advise against implementing feature extraction on your own. It's crucial to maintain consistency in feature extraction between the training environment and inference in {es}. By utilizing eland tooling, which is developed and tested in tandem with {es}, you can ensure that they function together consistently.
+
+* Feature extraction is performed by executing queries on the {es} server which could cause a lot of stress on your cluster, especially when your judgment list contains a lot of examples or you have many features. Our feature logger implementation is designed to minimize the number of search requests sent to the server in order to reduce the load, however building the training dataset might best be performed using an {es} cluster that is isolated from any user-facing, production traffic
+
+[discrete]
+[[learning-to-rank-model-deployment]]
+===== Deploy your model into {es}
+
+Once your model is trained you will be able to deploy it into your {es} cluster. For this purpose, eland provides the `MLModel.import_ltr_model method`:
+
+[source,python]
+----
+from eland.ml import MLModel
+
+LEARNING_TO_RANK_MODEL_ID="ltr-model-xgboost"
+
+MLModel.import_ltr_model(
+    es_client=es_client,
+    model=ranker,
+    model_id=LEARNING_TO_RANK_MODEL_ID,
+    ltr_model_config=ltr_config,
+    es_if_exists="replace",
+)
+----
+// NOTCONSOLE
+
+This method will serialize the trained model and the Learning To Rank configuration (including feature extraction) in a format that {es} can understand before sending it to Elasticsearch using the https://www.elastic.co/guide/en/elasticsearch/reference/current/put-trained-models.htmlp[Create Trained Models API].
```

Comment:
```suggestion
This method will serialize the trained model and the Learning To Rank configuration (including feature extraction) in a format that {es} can understand before sending it to Elasticsearch using the <<put-trained-models,Create Trained Models API>>.
```

Reply:
`.htmlp` is breaking CI, but we can also change this to internal link formatting